addons/l10n_mx/models/account.py

```diff
@@ -0,0 +1,75 @@
+# coding: utf-8
+# Copyright 2016 Vauxoo (https://www.vauxoo.com) <info@vauxoo.com>
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+import re
+from odoo import models, api, fields, _
+
+
+class AccountAccount(models.Model):
+    _inherit = 'account.account'
+
+    @api.multi
+    def assign_account_tag(self):
+        """Based on account code will be assigned by default the corresponding tag by tag code, this method is designed
+        to be called from a server action in order to ensure in the background the tag is being assigned.
+
+        The server action should be created like this:
+
+        ```
+        if object.company_id.country_id.id == env.ref('base.mx').id:
+            object.assign_account_tag()
+        ```
+        """
+        for account in self:
+            tags = account.search_tags()
+            account.write({'tag_ids': [(4, tags)]})
+
+    @api.multi
+    def search_tags(self):
+        self.ensure_one()
+        account_tag_obj = self.env['account.account.tag']
+        if not self.code:
+            return account_tag_obj
+        if not self.is_valid_account_number():
+            return account_tag_obj
+        account = self.get_account_tuple()
+        tag_parent = account_tag_obj.search([
+            ('name', 'ilike', str(account[0])),
+            ('color', '=', 1)])
+        tag = account_tag_obj.search([
+            ('name', 'ilike', '%s.%s' % (account[0], account[1])),
+            ('color', '=', 4)])
+        tags = tag | tag_parent | self.tag_ids.filtered(lambda r: not r.color in [4, 1])
+        return tags
+
+    @api.onchange('code')
+    def _onchange_code(self):
+        if self.company_id.country_id.id == self.env.ref('base.mx').id:
+            tags = self.search_tags()
+            self.tag_ids = tags
+
+    def search_regex(self):
+        """ Review if the given number is a valid account number
+        :return: the regex search None or regex object
+        """
+        return re.search(
+            '^(?P<first>[1-8][0-9][0-9])[,.]'
+            '(?P<second>[0-9][0-9])[,.]'
+            '(?P<third>[0-9]{2,3})$', self.code)
+
+    def is_valid_account_number(self):
+        """:return: Boolean """
+        return True if self.search_regex() else False
+
+    def get_account_tuple(self):
+        """:return: separate parts of the account"""
+        return self.search_regex().groups()
+
+
+class AccountAccountTag(models.Model):
+    _inherit = 'account.account.tag'
+
+    nature = fields.Selection([
+        ('D', _('Debitable Account')), ('A', _('Creditable Account'))],
```

Comment:
We decided use directly the value that we actually need in the xml:

![](http://screenshots.vauxoo.com/nhomar/Selection_102.png)

But I can left this english valid and convert afterwards (Are you agreed left this on this way? or do you prefer another approach?)